File: packages/singular/ios/Plugin/Classes/Options/TrackRevenueOptions.swift
import Foundation
import Capacitor

@objc public class TrackRevenueOptions: NSObject {
    let amount: Double
    let attributes: [String: Any]?
    let currency: String
    let eventName: String?

    init(_ call: CAPPluginCall) throws {
        guard let amount = call.getDouble("amount") else {
            throw CustomError.amountMissing
        }
        guard let currency = call.getString("currency") else {
            throw CustomError.currencyMissing
        }
        self.amount = amount
        self.currency = currency
        self.attributes = SingularHelper.createHashMapFromJSObject(call.getObject("attributes"))
        self.eventName = call.getString("eventName")
    }
}

File: packages/singular/android/src/main/java/io/capawesome/capacitorjs/plugins/singular/classes/options/TrackRevenueOptions.java
package io.capawesome.capacitorjs.plugins.singular.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.singular.SingularHelper;
import io.capawesome.capacitorjs.plugins.singular.classes.CustomExceptions;
import java.util.Map;

public class TrackRevenueOptions {

    private final double amount;

    @Nullable
    private final Map<String, Object> attributes;

    @NonNull
    private final String currency;

    @Nullable
    private final String eventName;

    public TrackRevenueOptions(@NonNull PluginCall call) throws Exception {
        Double amount = call.getDouble("amount");
        if (amount == null) {
            throw CustomExceptions.AMOUNT_MISSING;
        }
        String currency = call.getString("currency");
        if (currency == null) {
            throw CustomExceptions.CURRENCY_MISSING;
        }
        this.amount = amount;
        this.attributes = SingularHelper.createHashMapFromJSONObject(call.getObject("attributes"));
        this.currency = currency;
        this.eventName = call.getString("eventName");
    }

    public double getAmount() {
        return amount;
    }

    @Nullable
    public Map<String, Object> getAttributes() {
        return attributes;
    }

    @NonNull
    public String getCurrency() {
        return currency;
    }

    @Nullable
    public String getEventName() {
        return eventName;
    }
}
